internal/envoy/install_test.go

```diff
@@ -248,30 +244,20 @@ type installTest struct {
 }
 
 func setupInstallTest(t *testing.T) (*installTest, func()) {
-	var tearDown []func()
-
-	tempDir, removeTempDir := morerequire.RequireNewTempDir(t)
-	tearDown = append(tearDown, removeTempDir)
-
 	versionsServer := test.RequireEnvoyVersionsTestServer(t, version.LastKnownEnvoy)
-	tearDown = append(tearDown, versionsServer.Close)
-
+	homeDir := t.TempDir()
 	return &installTest{
-			ctx:        context.Background(),
-			tempDir:    tempDir,
-			tarballURL: test.TarballURL(versionsServer.URL, runtime.GOOS, runtime.GOARCH, version.LastKnownEnvoy),
-			GlobalOpts: globals.GlobalOpts{
-				HomeDir:          tempDir,
-				EnvoyVersionsURL: versionsServer.URL + "/envoy-versions.json",
-				Out:              new(bytes.Buffer),
-				Platform:         globals.DefaultPlatform,
-				RunOpts: globals.RunOpts{
-					EnvoyPath: filepath.Join(tempDir, "versions", string(version.LastKnownEnvoy), binEnvoy),
-				},
+		ctx:        context.Background(),
+		tempDir:    t.TempDir(),
```

Comment:
Oh is this change intentional? - before we pass `tempDir` both for here and HomeDir below, but not we use different dir.

Reply:
yeah it feels like an accident. I think before when we couldn't get on-demand directories without a cleanup function, probably led pressure to re-use directories. I undid that here to avoid that since it is easy now.